### PR TITLE
Authz uri builder scopes

### PR DESCRIPTION
**Pull request by sgryt**
Related to #3 but thorny enough to warrant its own PR.

### CODE REVIEW

File: src/components/AuthorizeURLBuilder.tsx [hunk 1]
```diff
@@ -13,7 +13,10 @@ interface AuthorizeURLOptions {
   response_mode: "query" | "fragment",
   acr_values: string[],
   acr_values_quirk: "none" | "login_hint" | "path",
-  nonce: string
+  nonce: string,
+  availableScopes : string[]
+  selectedScopes : string []
+  scopes_quirk : string
```
Comment (mickhansen) [inline src/components/AuthorizeURLBuilder.tsx:19]: set as `"none" | "login_hint"` instead of string

File: src/components/AuthorizeURLBuilder.tsx [hunk 2]
```diff
@@ -207,6 +263,46 @@ export default function AuthorizeURLBuilder() {
           </div>
         ))}
       </div>
+
+      {options.availableScopes.length > 0 ? (
+        <div>
+          <label>
```
Comment (mickhansen) [inline src/components/AuthorizeURLBuilder.tsx:269]: needs `className="block text-gray-700 text-sm font-bold mb-2"`

File: src/components/AuthorizeURLBuilder.tsx [hunk 3]
```diff
@@ -207,6 +263,46 @@ export default function AuthorizeURLBuilder() {
...
+              <small>
+                Some integrations, like Auth0, require that you pass scopes through the login_hint.<br />
+              </small>
+            </div>
+          </div>
+        </div>
+      ) : null}
       
       {options.acr_values.length == 1 ? (
         <div className="mb-4 grid grid-cols-2 gap-4">
```
Comment (mickhansen) [inline src/components/AuthorizeURLBuilder.tsx:308]: scopes, scopes quirk handling and acr values handling should probably all be in the same grid so they can take up some horizontal space

**Comment by sgryt** [comment]
@mickhansen Updated

**Comment by mickhansen** [comment]
LGTM

**Comment by mickhansen** [comment]
@sgryt Ready for merge?

**Comment by sgryt** [comment]
> @sgryt Ready for merge?

[same as heer](https://github.com/criipto/document-signatures-docs/pull/3#issuecomment-1094842730)

**Pull request merged**